jrnl/cli.py

```diff
@@ -478,5 +478,8 @@ def run(manual_args=None):
         journal.write()
 
     elif args.delete:
+        other_entries = [e for e in old_entries if e not in journal.entries]
```

Comment:
We try to avoid looping through entire journals as much as possible. This can be slow for very large journals (you'd be surprised how large some user's journals are).